packages/worker/src/sdk/users/users.ts

```diff
@@ -203,6 +228,131 @@ export const save = async (
   }
 }
 
+export const addTenant = async (
+  tenantId: string,
+  _id: string,
+  email: string
+) => {
+  if (env.MULTI_TENANCY) {
+    const afterCreateTenant = () =>
+      migrations.backPopulateMigrations({
+        type: MigrationType.GLOBAL,
+        tenantId,
+      })
+    await tenancy.tryAddTenant(tenantId, _id, email, afterCreateTenant)
+  }
+}
+
+export const bulkCreate = async (newUsersRequested: any[], groups: any) => {
+  const db = tenancy.getGlobalDB()
+  const tenantId = tenancy.getTenantId()
+
+  let usersToSave: any[] = []
+  let newUsers: any[] = []
+
+  const allUsers = await db.allDocs(
+    dbUtils.getGlobalUserParams(null, {
+      include_docs: true,
+    })
+  )
+  let mapped = allUsers.rows.map((row: any) => row.id)
+
+  const currentUserEmails = mapped.map((x: any) => x.email) || []
+  for (const newUser of newUsersRequested) {
+    if (
+      newUsers.find((x: any) => x.email === newUser.email) ||
+      currentUserEmails.includes(newUser.email)
+    ) {
+      continue
+    }
+    newUser.userGroups = groups
+    newUsers.push(newUser)
+  }
+
+  // Figure out how many builders we are adding and create the promises
+  // array that will be called by bulkDocs
+  let builderCount = 0
+  newUsers.forEach((user: any) => {
+    if (eventHelpers.isAddingBuilder(user, null)) {
+      builderCount++
+    }
+    usersToSave.push(
+      buildUser(
+        user,
+        {
+          hashPassword: true,
+          requirePassword: user.requirePassword,
+          bulkCreate: false,
+        },
+        tenantId
+      )
+    )
+  })
+
+  const usersToBulkSave = await Promise.all(usersToSave)
+  await quotas.addDevelopers(() => db.bulkDocs(usersToBulkSave), builderCount)
```

Comment:
Are there changes to push to the pro repo around this? it's `addDeveloper` currently without the `s`

Reply:
yep, pushing the pro changes up shortly